workers/unity/Packages/com.improbable.gdk.tools/ScriptableGdkToolsConfiguration.cs

```diff
@@ -0,0 +1,176 @@
+        public string CodegenOutputDir = DefaultValues.CodegenOutputDir;
+
+        public const string AssetPath = "Assets/Config/GdkToolsConfiguration.asset";
+
+        private const string DownloadCoreSdkLabel = "Core SDK Options";
+        private const string SchemaStdLibDirLabel = "Schema Standard Library Directory";
+
+        private const string CodeGeneratorLabel = "Code Generator Options";
+        private const string CodegenOutputDirLabel = "Code Generator Output Directory";
+        private const string SchemaSourceDirsLabel = "Schema Source Directories";
+
+        private const string AddSchemaDirButtonText = "Add Schema Source Directory";
+        private const string RemoveSchemaDirButtonText = "Remove";
+
+        private const string SaveConfigurationButtonText = "Save GDK Tools Configuration";
+        private const string ResetConfigurationButtonText = "Reset GDK Tools Configuration to Default";
+
+        private Vector2 scrollPosition = Vector2.zero;
+
+        public void OnGUI()
+        {
+            scrollPosition = GUILayout.BeginScrollView(scrollPosition);
+
+            GUILayout.Label(DownloadCoreSdkLabel);
+            GUILayout.Space(5);
+            GUILayout.Label(SchemaStdLibDirLabel);
+            SchemaStdLibDir = GUILayout.TextField(SchemaStdLibDir);
+
+            GUILayout.Space(10);
+            EditorGUILayout.TextArea("", GUI.skin.horizontalSlider);
+            GUILayout.Space(10);
+
+            GUILayout.Label(CodeGeneratorLabel);
+            GUILayout.Space(5);
+            GUILayout.Label(CodegenOutputDirLabel);
+            CodegenOutputDir = GUILayout.TextField(CodegenOutputDir);
+
+            GUILayout.Space(5);
+            GUILayout.Label(SchemaSourceDirsLabel);
+            for (var i = 0; i < SchemaSourceDirs.Count; i++)
+            {
+                GUILayout.BeginHorizontal();
+                SchemaSourceDirs[i] = GUILayout.TextField(SchemaSourceDirs[i]);
+
+                if (GUILayout.Button(RemoveSchemaDirButtonText, GUILayout.Width(100)))
+                {
+                    SchemaSourceDirs.RemoveAt(i);
+                }
+
+                GUILayout.EndHorizontal();
+            }
+
+            if (GUILayout.Button(AddSchemaDirButtonText, GUILayout.Width(250)))
+            {
+                SchemaSourceDirs.Add("");
+            }
+
+            GUILayout.Space(10);
+            EditorGUILayout.TextArea("", GUI.skin.horizontalSlider);
+            GUILayout.Space(10);
+
+            if (GUILayout.Button(SaveConfigurationButtonText, GUILayout.Width(250)))
+            {
+                Save();
+            }
+
+            GUILayout.Space(15);
+
+            if (GUILayout.Button(ResetConfigurationButtonText, GUILayout.Width(250)))
+            {
+                ResetToDefault();
+            }
+
+            GUILayout.EndScrollView();
+        }
+
+        private void Save()
```

Comment:
This becomes unnecessary if you use the built-in Inspector + Unity serialization